### PR TITLE
Extended authenticated route with requiredRole property

### LINKED ISSUE #260: Frontend: Extend AuthenticateRoute with a role property
Add a new property `requiredRole` of type string which, if it is provided, will check that the `userRole` coming from `useUser` matches that role.

### PR DESCRIPTION
**Pull request by AnilMula**
# Description

This PR is to add a new prop "requiredRole" to the AuthenticateRoute component

Fixes #260 

# How to test?

Not tested

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/App.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ function App() {
           <EventsPage />
         </AuthenticatedRoute>
 
-        <AuthenticatedRoute exact path="/profile">
+        <AuthenticatedRoute exact path="/profile" requiredRole="student">
```
Comment (tpetrina) [inline src/client/App.js]: Remove this as all roles can view their profile.

File: src/client/components/Auth/AuthenticatedRoute.js [hunk 2]
```diff
@@ -25,10 +25,10 @@ function AuthenticatedRoute({ children, requiredRole, ...rest }) {
           }
           // I am just adding here but i am not sure whether it is the right place
           if (userRole === requiredRole) {
-            return 'matches'; // no idea on what to do if it matches
+            return children; // no idea on what to do if it matches
           }
 
-          return children;
+          return <ErrorPage />;
```
Comment (tpetrina) [inline src/client/components/Auth/AuthenticatedRoute.js:30]: Excellent!

File: src/client/components/Auth/AuthenticatedRoute.js [hunk 3]
```diff
@@ -25,10 +25,10 @@ function AuthenticatedRoute({ children, requiredRole, ...rest }) {
           }
           // I am just adding here but i am not sure whether it is the right place
           if (userRole === requiredRole) {
-            return 'matches'; // no idea on what to do if it matches
+            return children; // no idea on what to do if it matches
           }
```
Comment (tpetrina) [inline src/client/components/Auth/AuthenticatedRoute.js]: Remove comments.
Have you tested what happens if you remove the code in `App.js` and still try to open this page?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Almost there! Small tweaks required.

File: src/client/components/Auth/AuthenticatedRoute.js (hunk 3 above)
Comment (AnilMula) [reply src/client/components/Auth/AuthenticatedRoute.js]: it shows 404.

Comment (tpetrina) [reply src/client/components/Auth/AuthenticatedRoute.js]: So it is obviously wrong. What should happen if there is no `requiredRole`?

Comment (AnilMula) [reply src/client/components/Auth/AuthenticatedRoute.js]: oh ya , it should not display 404, i changed

**Review by tpetrina — APPROVED** [review]
Super stuff!

**Pull request merged**